src/core/src/collection.rs

```diff
@@ -215,6 +215,14 @@ impl Collection {
         assert_eq!(sig.signatures.len(), 1);
         Ok(sig)
     }
+
+    pub fn intersect_manifest(&self, mf: &Manifest) -> Self {
```

Comment:
would it be simpler (or complementary) to enable intersect of the collections directly i.e. pass a ref to the Collection (`other`) and then use `self.manifest.intersect(other.manifest)`?

I guess it probably doesn't matter as long as we're passing in a reference to either the collection or manifest...

Reply:
this comes directly from the `MultiCollection` code use case, where we read a manifest in directly; would be frustrating to have to create a `Collection` first.

we might be able to coerce a `Collection` into a `Manifest`, though, which would make it work more smoothly when you do have a `Collection`... I'll look!